daml_dit_if/main/integration_deferral_queue.py

```diff
@@ -0,0 +1,34 @@
+import asyncio
+
+from typing import Callable
+
+from .log import LOG
+
+
+DeferredAction = Callable[[], None]
+
+class IntegrationDeferralQueue:
+
+    def __init__(self):
+        self.queue = \
+            asyncio.Queue(maxsize=10)  # type: asyncio.Queue[DeferredAction]
+
+    async def put(self, action: DeferredAction):
+        await self.queue.put(action)
+
+    def qsize(self):
+        return self.queue.qsize()
+
+    async def start(self):
+        LOG.debug('Deferral worker starting.')
+
+        while True:
```

Comment:
So I guess this never needs to terminate naturally, since this is meant to be part of a long-running process?

Reply:
Yup